TestApplication/SUTestApplicationDelegate.m

```diff
@@ -0,0 +1,184 @@
+        NSAlert *alreadyUpdatedAlert = [[NSAlert alloc] init];
+        alreadyUpdatedAlert.messageText = @"Update succeeded!";
+        alreadyUpdatedAlert.informativeText = @"This is the updated version of Sparkle Test App.\n\nDelete and rebuild the app to test updates again.";
+        [alreadyUpdatedAlert runModal];
+        
+        [[NSApplication sharedApplication] terminate:nil];
+    }
+    
+    NSFileManager *fileManager = [[NSFileManager alloc] init];
+    
+    // Locate user's cache directory
+    NSError *cacheError = nil;
+    NSURL *cacheDirectoryURL = [fileManager URLForDirectory:NSCachesDirectory inDomain:NSUserDomainMask appropriateForURL:nil create:YES error:&cacheError];
+    
+    if (cacheDirectoryURL == nil) {
+        NSLog(@"Failed to locate cache directory with error: %@", cacheError);
+        assert(cacheDirectoryURL != nil);
+    }
+    
+    NSString *bundleIdentifier = mainBundle.bundleIdentifier;
+    assert(bundleIdentifier != nil);
+    
+    
+    // Create a directory that'll be used for our web server listing
+    NSURL *serverDirectoryURL = [cacheDirectoryURL URLByAppendingPathComponent:bundleIdentifier];
+    if ([fileManager fileExistsAtPath:serverDirectoryURL.path]) {
+        NSError *removeServerDirectoryError = nil;
+        
+        if (![fileManager removeItemAtURL:serverDirectoryURL error:&removeServerDirectoryError]) {
+            assert(NO);
+        }
+    }
+    
+    NSError *createDirectoryError = nil;
+    if (![fileManager createDirectoryAtURL:serverDirectoryURL withIntermediateDirectories:YES attributes:nil error:&createDirectoryError]) {
+        NSLog(@"Failed creating directory at %@ with error %@", serverDirectoryURL.path, createDirectoryError);
+        assert(createDirectoryError != nil);
+    }
+    
+    NSURL *bundleURL = mainBundle.bundleURL;
+    assert(bundleURL != nil);
+    
+    // Copy main bundle into server directory
+    NSURL *destinationBundleURL = [serverDirectoryURL URLByAppendingPathComponent:bundleURL.lastPathComponent];
+    NSError *copyBundleError = nil;
+    if (![fileManager copyItemAtURL:bundleURL toURL:destinationBundleURL error:&copyBundleError]) {
+        NSLog(@"Failed to copy main bundle into server directory with error %@", copyBundleError);
+        assert(NO);
+    }
+    
+    // Update bundle's version keys to latest version
+    NSURL *infoURL = [[destinationBundleURL URLByAppendingPathComponent:@"Contents"] URLByAppendingPathComponent:@"Info.plist"];
+    
+    assert([fileManager fileExistsAtPath:infoURL.path]);
+    
+    NSMutableDictionary *infoDictionary = [[NSMutableDictionary alloc] initWithContentsOfURL:infoURL];
+    infoDictionary[(NSString *)kCFBundleVersionKey] = UPDATED_VERSION;
+    infoDictionary[@"CFBundleShortVersionString"] = UPDATED_VERSION;
+    
+    assert([infoDictionary writeToURL:infoURL atomically:NO]);
```

Comment:
Assertions should not be used for important side effects, as such disabling them should have no effect on the execution of the program. Instead of:

    assert([infoDictionary writeToURL:infoURL atomically:NO]);

Use:

    BOOL ok = [infoDictionary writeToURL:infoURL atomically:NO];
    assert(ok);